active/NavigationView/HierarchicalNav/HierarchicalNavigationView.md

```diff
@@ -0,0 +1,265 @@
+
+# Background
+The Xaml [NavigationView](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.NavigationView) control enables navigation of app content. It has a header, a view for the main content, and a menu pane for navigation commands. There is additional discussion about how to use this control in the [Navigation view](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/navigationview) design document.
+
+Currently, NavigationView's MenuItems list allows for displaying a flat list of items in the pane. 
+It's common for apps to want to present users with a hierarchical navigation tree. 
+This feature adds the capability to nest items within the pane.
+
+# Hierarchy
+To show a hierarchical list of nested navigation items in the pane, use NavigationViewItem's `MenuItems` property.
+Each *NavigationViewItem* can contain other NavigationViewItems and organizing elements like item headers and separators. 
+Although NavigationViewItem can contain any number of nested levels, we recommend keeping your app’s navigation hierarchy shallow. 
+We believe two levels is ideal for usability and comprehension.
+
+NavigationView shows hierarchy in all its pane display modes, including Top and LeftCompact.
+
+*Note:* The following is a comp and will be updated with screenshots before this doc is published.
+![NavigationView in Left, LeftCompact, and Top modes showing hierarchy](NavigationView_Hierarchy.png)
+
+## Selection
+By default, any item can contain children, be invoked, or be selected.
+There's more detail about the [ItemInvoked](https://docs.microsoft.com/en-us/uwp/api/microsoft.ui.xaml.controls.navigationview.iteminvoked?view=winui-2.3) and [SelectionChanged](https://docs.microsoft.com/en-us/uwp/api/microsoft.ui.xaml.controls.navigationview.selectionchanged?view=winui-2.3) events later in this document.
+When providing users with a hierarchical tree of navigation options, you may choose to make parent items non-selectable, for example when your app doesn't have a destination page associated with that parent item.
+To prevent an item from showing the selection indicator when invoked, set its [SelectsOnInvoked](https://docs.microsoft.com/en-us/uwp/api/microsoft.ui.xaml.controls.navigationviewitem.selectsoninvoked?view=winui-2.3) property to False.
+
+Selected items will draw their selection indicators along their left edge when in left mode or their bottom edge when in top mode. 
+The selected item may not always remain visible.
+For example, the selected item may be a child node inside a non-expanded subtree.
+In this situation, the first visible ancestor of the selected item will show as selected, and the selection indicator will move as users expand the subtree. 
+The entire navigation view will show no more than one selection indicator.
+
+## Keyboarding
```

Comment:
Is this going to use TreeView as a source or is this implementing the TreeView's behavior but is a copy? I ask because we did have a tons of discussion about how to support gamepad with TreeView and I'm wondering if this is something you should also keep in mind and support here?

Reply:
It's a copy. I admit I had not given gamepad any thought...

Reply:
It is not using TreeView and it is not a copy either. It is a completely different implementation using ItemsRepeater compared to TreeView which uses ListView with a flattened tree.

Reply:
Sorry for being vague, I meant the behavior is being copied. 

The only new concept we're introducing is the use of left/right arrow keys (which do nothing in NavigationView today). Those are present on the controller as well. What kind of differences between controller and keyboard do I really need to think about in this case?

Reply:
There's quite a bit of work that happened. Unfortunately Premalini has the details and I don't want to be found quoting on anything wrong, but for example, we use A/B buttons and not directional keys to open/collapse the items (I think... it's been such a long time and went back and forth...)

Reply:
@predavid can you help fill in the details?
I would expect the A button to invoke the item rather than expand.

Reply:
@ojhad Can you provide details on the gamepad behavior that you currently have ? @predavid can you comment if the current behavior matches your expectation (relative to TreeView) ?

Reply:
Currently, with the existing gamepad behavior, in order to open the flyout you need to invoke the item using the A button. Directional keys are just used for x/y navigation.